yt_dlp/extractor/peertube.py

```diff
@@ -1369,3 +1369,60 @@ def _real_extract(self, url):
         return self.playlist_result(
             entries, playlist_id, playlist_title, playlist_description,
             timestamp=playlist_timestamp, channel=channel, channel_id=channel_id, thumbnail=thumbnail)
+
+
+class PeerTubeChannelIE(InfoExtractor):
+    IE_NAME = 'PeerTube:Channel'
+    _VALID_URL = r'(?x)https?://(?P<host>%s)/(?P<type>[ac])/(?P<id>[^/]+)/(videos|video-channels)' % PeerTubeIE._INSTANCES_RE
+    _API_BASE = 'https://%s/api/v1/%s/%s%s'
+    _TESTS = [{
+        'url': 'https://peertube2.cpy.re/a/chocobozzz/videos',
+        'info_dict': {
+            'id': 'chocobozzz',
+            'timestamp': 1553874564,
+            'title': 'chocobozzz',
+        },
+        'playlist_mincount': 2,
+    }, {
+        'url': 'https://framatube.org/c/bf54d359-cfad-4935-9d45-9d6be93f63e8/videos',
+        'info_dict': {
+            'id': 'bf54d359-cfad-4935-9d45-9d6be93f63e8',
+            'timestamp': 1519917377,
+            'title': 'Les vidéos de Framasoft',
+        },
+        'playlist_mincount': 345,
+    }]
+    _PAGE_SIZE = 30
+
+    def _call_api(self, host, name, path, base, note=None, **kwargs):
+        return self._download_json(
+            self._API_BASE % (host, base, name, path), name,
+            note=note, **kwargs)
+
+    def _fetch_page(self, host, uuid, type, page):
+        page += 1
+        video_data = self._call_api(
+            host, uuid, f'/videos?sort=-createdAt&start={self._PAGE_SIZE * (page - 1)}&count={self._PAGE_SIZE}&nsfw=both',
+            type, note=f'Downloading page {page}').get('data', {})
```

Comment:
here aswell, it should be list and not dict
doing for each on dict gives you the keys.